pkg/helpers/crd.go

```diff
@@ -54,22 +57,22 @@ func CreateOrUpdateGitRepo(obj metav1.Object, gvk schema.GroupVersionKind, templ
 		if err != nil {
 			return fmt.Errorf("could not update existing repo: %v", err)
 		}
+		existingRepo.Spec = repo.Spec
 
-		if !reflect.DeepEqual(existingRepo.Spec, repo.Spec) {
-			existingRepo.Spec = repo.Spec
+		err = client.Update(context.TODO(), existingRepo)
```

Comment:
Is the update really necessary? Can't we just leave this to whoever is calling this func?

Reply:
Not really sure what the benefit would be by removing the actual update from `CreateOrUpdateGitRepo`. I feel like that would clutter the tenant and cluster reconciles with unnecessary GitRepo code. Also if we implement another CR that needs a GitRepo we'd have to remember to call update after calling `CreateOrUpdateGitRepo`. And removing it here would mean we'd have to add it to two places so we'd have code duplication.

Reply:
Sorry, my bad I mixed up the CRs.

Reply:
No problem :)